### ISSUE TITLE
Dual Monitors not working?

### ISSUE THREAD

**Issue by georgearnall** [xythobuz/DisplayBacklight#1]
I've tried to configure you're app but I can only get my strip to work on a single monitor, all of the LEDs are working.

I have 2x 1080p monitors and my MBP running. It should only be configured for the external displays. Picture: https://cl.ly/1f3y2Z2A452T
152 leds, 16 vertical, 30 horizontal per display
```
struct DisplayAssignment displays[] = {
    { 1920, 1080 },
    { 1920, 1080 }
};

struct LEDStrand strands[] = {
 //ledID, len, dispID,startX,startY, dir, division
    {   0, 30, 0, 1920, 1080,  DIR_LEFT, 1920 / 30 },
    {  30, 16, 0,    0, 1080,    DIR_UP, 1080 / 16 },
    {  46, 30, 0,    0,    0, DIR_RIGHT, 1920 / 30 },
    {  76, 30, 1,    0,    0, DIR_RIGHT, 1920 / 30 },
    { 106, 16, 1, 1920,    0,  DIR_DOWN, 1080 / 16 },
    { 122, 30, 1, 1920, 1080,  DIR_LEFT, 1920 / 30 }
};
```

**Comment by xythobuz** [comment]
Well, it's not actually supported to use these with monitors that have the same resolution, currently... :laughing: 

This is hinted at by the comment just above the part you're quoting:

    // Identify your displays here. Currently they're only distinguished by their resolution.

If you take a look at the code snippet [here](https://github.com/xythobuz/DisplayBacklight/blob/master/DisplayBacklight/AppDelegate.m#L519):

    // Try to find the matching display id for the strand associations
    for (int n = 0; n < (sizeof(displays) / sizeof(displays[0])); n++) {
        if ((width == displays[n].width) && (height == displays[n].height)) {
            // ...
        }
    }

You can probably see easily why this won't work for your case...

The thing is, I'm no longer running Mac OS X on my Desktop, I've finally moved on to Linux. So I'm sadly not really able to implement and test this for you. But the general idea is:

 - find some other way of distinguishing the displays. Preferrably some unique ID or serial number that is provided by the Cocoa API or whatever
 - properly change the code to now use this as the new index for the display settings (of course this is easier said than done)

If you feel you're up to the task, please, go ahead :smiley: I'd be glad to assist in any way I can...

**Comment by xythobuz** [comment]
Yeah, there's a lot of redundant information in the currently implemented configuration. But the user will at least need to be able to configure the offset and length of each strip in leds and pixels, otherwise the colours won't necessarily match up. So you will still have _some_ user configurable options. But all the stuff like display IDs and resolutions is available from the API, I think.
